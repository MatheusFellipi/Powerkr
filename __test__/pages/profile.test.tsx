import { render } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import Profile from "../../src/app/(tabs)/profile";
import config from "../../tamagui.config";

describe("<Profile />", () => {
  it("renders correctly", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <Profile />
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
