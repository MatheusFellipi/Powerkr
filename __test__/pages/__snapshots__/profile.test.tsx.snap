// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders correctly 1`] = `
<View
  style={
    {
      "paddingLeft": 18,
      "paddingRight": 18,
      "paddingTop": 60,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <View
      style={
        [
          {
            "position": "absolute",
          },
          {
            "left": 0,
          },
        ]
      }
    >
      <SvgMock />
    </View>
    <Text
      fontFamily="Inter"
      style={
        {
          "color": "#fff",
          "fontFamily": "Inter",
          "fontSize": 20,
        }
      }
      suppressHighlighting={true}
    >
      Perfil
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "marginBottom": 18,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomLeftRadius": 100000,
          "borderBottomRightRadius": 100000,
          "borderTopLeftRadius": 100000,
          "borderTopRightRadius": 100000,
          "flexDirection": "column",
          "height": 204,
          "justifyContent": "center",
          "marginBottom": 18,
          "marginTop": 24,
          "maxHeight": 204,
          "maxWidth": 204,
          "minHeight": 204,
          "minWidth": 204,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "relative",
          "width": 204,
        }
      }
    >
      <View
        style={
          {
            "bottom": 0,
            "flexDirection": "column",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      >
        <Image
          accessibilityLabel="Cam"
          fullscreen={true}
          onError={[Function]}
          onLoad={[Function]}
          style={
            {
              "height": 204,
              "width": 204,
            }
          }
        />
      </View>
      <View
        style={
          {
            "backgroundColor": "hsl(216, 50.0%, 11.8%)",
            "bottom": 0,
            "flexDirection": "column",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
    <Text
      fontFamily="Inter"
      style={
        {
          "color": "#fff",
          "fontFamily": "Inter",
          "fontSize": 20,
        }
      }
      suppressHighlighting={true}
    />
    <Text
      fontFamily="Inter"
      style={
        {
          "color": "#9AA4B2",
          "fontFamily": "Inter",
          "fontSize": 14,
        }
      }
      suppressHighlighting={true}
    />
  </View>
  <View
    style={
      {
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          fontFamily="Inter"
          style={
            {
              "color": "#CDD5DF",
              "fontFamily": "Inter",
              "fontSize": 14,
            }
          }
          suppressHighlighting={true}
        >
          Nome
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            fontFamily="Inter"
            style={
              {
                "color": "#9AA4B2",
                "fontFamily": "Inter",
                "fontSize": 14,
                "marginRight": 24,
              }
            }
            suppressHighlighting={true}
          >
            Nome Completo
          </Text>
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "height": 21,
          "minHeight": 21,
          "minWidth": 21,
          "width": 21,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          fontFamily="Inter"
          style={
            {
              "color": "#CDD5DF",
              "fontFamily": "Inter",
              "fontSize": 14,
            }
          }
          suppressHighlighting={true}
        >
          E-mail
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            fontFamily="Inter"
            style={
              {
                "color": "#9AA4B2",
                "fontFamily": "Inter",
                "fontSize": 14,
                "marginRight": 24,
              }
            }
            suppressHighlighting={true}
          >
            nome.sobrenome@gmail.com
          </Text>
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "height": 21,
          "minHeight": 21,
          "minWidth": 21,
          "width": 21,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          fontFamily="Inter"
          style={
            {
              "color": "#CDD5DF",
              "fontFamily": "Inter",
              "fontSize": 14,
            }
          }
          suppressHighlighting={true}
        >
          Alterar senha
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            fontFamily="Inter"
            style={
              {
                "color": "#9AA4B2",
                "fontFamily": "Inter",
                "fontSize": 14,
                "marginRight": 24,
              }
            }
            suppressHighlighting={true}
          />
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "height": 21,
          "minHeight": 21,
          "minWidth": 21,
          "width": 21,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        fontFamily="Inter"
        style={
          {
            "color": "#CDD5DF",
            "fontFamily": "Inter",
            "fontSize": 14,
          }
        }
        suppressHighlighting={true}
      >
        Notificações
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#F7B519"
        style={
          [
            {
              "height": 31,
              "width": 51,
            },
            {
              "backgroundColor": "#3e3e3e",
              "borderRadius": 16,
            },
          ]
        }
        thumbTintColor="#f4f3f4"
        tintColor="#767577"
        value={true}
      />
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "height": 21,
          "minHeight": 21,
          "minWidth": 21,
          "width": 21,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        fontFamily="Inter"
        style={
          {
            "color": "#CDD5DF",
            "fontFamily": "Inter",
            "fontSize": 14,
          }
        }
        suppressHighlighting={true}
      >
        Modo Noturno
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#F7B519"
        style={
          [
            {
              "height": 31,
              "width": 51,
            },
            {
              "backgroundColor": "#3e3e3e",
              "borderRadius": 16,
            },
          ]
        }
        thumbTintColor="#f4f3f4"
        tintColor="#767577"
        value={true}
      />
    </View>
    <View
      pointerEvents="none"
      style={
        {
          "height": 21,
          "minHeight": 21,
          "minWidth": 21,
          "width": 21,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "column",
        }
      }
    >
      <View
        accessibilityRole="button"
        cursor="pointer"
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#202939",
            "borderBottomColor": "#4B5565",
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderBottomWidth": 1,
            "borderLeftColor": "#4B5565",
            "borderLeftWidth": 1,
            "borderRightColor": "#4B5565",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "#4B5565",
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 44,
            "justifyContent": "center",
            "paddingLeft": 18,
            "paddingRight": 18,
          }
        }
      >
        <SvgMock
          color="#fff"
          height={20}
          size={14}
          width={20}
        />
        <View
          pointerEvents="none"
          style={
            {
              "height": 0,
              "minHeight": 0,
              "minWidth": 9.24,
              "width": 9.24,
            }
          }
        />
        <Text
          cursor="pointer"
          fontFamily="InterBold"
          lineBreakMode="clip"
          numberOfLines={1}
          style={
            {
              "color": "#fff",
              "flexGrow": 0,
              "flexShrink": 1,
              "fontFamily": "InterBold",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 23,
            }
          }
          suppressHighlighting={true}
          userSelect="none"
        >
          Ajuda
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          {
            "height": 24,
            "minHeight": 24,
            "minWidth": 24,
            "width": 24,
          }
        }
      />
      <View
        accessibilityRole="button"
        cursor="pointer"
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#202939",
            "borderBottomColor": "#4B5565",
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderBottomWidth": 1,
            "borderLeftColor": "#4B5565",
            "borderLeftWidth": 1,
            "borderRightColor": "#4B5565",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "#4B5565",
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 44,
            "justifyContent": "center",
            "paddingLeft": 18,
            "paddingRight": 18,
          }
        }
      >
        <SvgMock
          color="#fff"
          height={20}
          size={14}
          width={20}
        />
        <View
          pointerEvents="none"
          style={
            {
              "height": 0,
              "minHeight": 0,
              "minWidth": 9.24,
              "width": 9.24,
            }
          }
        />
        <Text
          cursor="pointer"
          fontFamily="InterBold"
          lineBreakMode="clip"
          numberOfLines={1}
          style={
            {
              "color": "#fff",
              "flexGrow": 0,
              "flexShrink": 1,
              "fontFamily": "InterBold",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 23,
            }
          }
          suppressHighlighting={true}
          userSelect="none"
        >
          Sair
        </Text>
      </View>
    </View>
  </View>
</View>
`;
